go/src/koding/klient/machine/machinegroup/idset/idset_test.go

```diff
@@ -0,0 +1,159 @@
+package idset
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	"koding/klient/machine"
+)
+
+func TestUnion(t *testing.T) {
+	tests := map[string]struct {
+		A, B, Expected []machine.ID
+	}{
+		"simple union": {
+			A:        genIDs("x1, x2, x3, x4"),
```

Comment:
I found that approach more readable and it was easier to modify when I replaced `[]machine.ID` with `machine.IDSlice` (see #10054). It's a matter of taste, If you want I can change it.